src/cargo/core/compiler/build_context/mod.rs

```diff
@@ -52,7 +52,7 @@ impl<'a, 'cfg> BuildContext<'a, 'cfg> {
         extra_compiler_args: HashMap<Unit<'a>, Vec<String>>,
     ) -> CargoResult<BuildContext<'a, 'cfg>> {
         let mut rustc = config.load_global_rustc(Some(ws))?;
-        if let Some(wrapper) = &build_config.rustc_wrapper {
+        if let Some(wrapper) = &build_config.primary_unit_rustc {
```

Comment:
Ah, remove these three lines.  The choice of whether or not to use `primary_unit_rustc` is done elsewhere.